Library/Homebrew/utils/shell.rb

```diff
@@ -98,17 +98,20 @@ def prepend_path_in_profile(path)
       end
     end
 
-    SHELL_PROFILE_MAP = {
-      bash: "~/.profile",
-      csh:  "~/.cshrc",
-      fish: "~/.config/fish/config.fish",
-      ksh:  "~/.kshrc",
-      mksh: "~/.kshrc",
-      rc:   "~/.rcrc",
-      sh:   "~/.profile",
-      tcsh: "~/.tcshrc",
-      zsh:  "~/.zshrc",
-    }.freeze
+    SHELL_PROFILE_MAP = T.let(
+      {
+        bash: "~/.profile",
+        csh:  "~/.cshrc",
+        fish: "~/.config/fish/config.fish",
+        ksh:  "~/.kshrc",
+        mksh: "~/.kshrc",
+        rc:   "~/.rcrc",
+        sh:   "~/.profile",
+        tcsh: "~/.tcshrc",
+        zsh:  "~/.zshrc",
+      }.freeze,
+      T::Hash[T.nilable(Symbol), String],
```

Comment:
```suggestion
      T::Hash[Symbol, String],
```
feels more appropriate?

Reply:
Tried that, but in `def profile`’s `SHELL_PROFILE_MAP.fetch(preferred, "~/.profile") `, `preferred` can be `nil` and there’s no default fallback.

Reply:
Does that produce a type-error? Feels like it might be nicer in that case to have an early `return` rather than attempting a `fetch` with `nil`. Thoughts?

Reply:
Yes. And good idea, yeah, I'll do that.

Reply:
Done!